src/main/webapp/app/course/manage/course-management-exercises.component.html

```diff
@@ -1,33 +1,99 @@
 <ng-container *ngIf="course">
-    <h3>
-        <span
-            >{{ course.title }} - {{ quizExercisesCount + programmingExercisesCount + modelingExercisesCount + fileUploadExercisesCount + textExercisesCount }}
-            {{ 'artemisApp.course.exercises' | artemisTranslate }}</span
-        >
-    </h3>
+    <div class="row">
+        <div class="col">
+            <h3>
+                <span>
+                    {{ course.title }}
+                    - {{ quizExercisesCount + programmingExercisesCount + modelingExercisesCount + fileUploadExercisesCount + textExercisesCount }}
+                    {{ 'artemisApp.course.exercises' | artemisTranslate }}
+                </span>
+            </h3>
+        </div>
+        <div class="col text-end align-self-center">
+            <button id="toggleSearchButton" class="btn btn-primary" type="button" (click)="toggleSearch()">
+                {{ 'artemisApp.course.search.toggleButton' | artemisTranslate }}
+            </button>
+        </div>
+    </div>
+
+    <jhi-course-management-exercises-search *ngIf="showSearch" (exerciseFilter)="exerciseFilter = $event"></jhi-course-management-exercises-search>
+    <div
+        *ngIf="
+            showSearch &&
+            filteredProgrammingExercisesCount + filteredQuizExercisesCount + filteredModelingExercisesCount + filteredTextExercisesCount + filteredFileUploadExercisesCount == 0 &&
+            !exerciseFilter.isEmpty()
+        "
+        class="alert alert-secondary"
+        role="alert"
+    >
+        {{ 'artemisApp.course.search.noResults' | artemisTranslate }}
+    </div>
     <jhi-alert></jhi-alert>
-    <jhi-course-exercise-card headingJhiTranslate="artemisApp.course.programmingExercises">
+
+    <jhi-course-exercise-card [hidden]="!['all', 'programming'].includes(exerciseFilter.exerciseTypeSearch)" headingJhiTranslate="artemisApp.course.programmingExercises">
```

Comment:
I think this `hidden` logic could also be moved to component and replaced with something like: `[hidden]="!isProgrammingExerciseType && !isAllExerciseType`

similar idea for the quiz, modelling, text, file-upload exercises.